pkg/speaker/subnet.go

```diff
@@ -348,11 +348,17 @@ func getNextHopFromPathAttributes(attrs []bgp.PathAttributeInterface) net.IP {
 	return nil
 }
 
-func getNextHopAttribute(neighborAddress, routeID string) string {
-	nextHop := routeID
+func (c *Controller) getNextHopAttribute(neighborAddress string) string {
```

Comment:
I mean this function should be like this:

```go
func (c *Controller) getNextHopAttribute(neighborAddress string) string {
	proto := util.CheckProtocol(neighborAddress)
	if nodeIP := c.config.NodeIPs[proto]; nodeIP != nil {
		return nodeIP.String()
	}

	nextHop := c.config.RouterID
	routes, err := netlink.RouteGet(net.ParseIP(neighborAddress))
	if err == nil && len(routes) == 1 && routes[0].Src != nil {
		nextHop = routes[0].Src.String()
	}
	return nextHop
}
```

Reply:
what if the route has a different source from the podIP? I meant, we should only use nodeIP when the nextHop equals podIP.